src/coreComponents/constitutive/fluid/PVTFunctions/PVTFunctionHelpers.hpp

```diff
@@ -0,0 +1,162 @@
+ * ------------------------------------------------------------------------------------------------------------
+ */
+
+/**
+ * @file PVTFunctionHelpers.hpp
+ */
+
+#ifndef GEOSX_CONSTITUTIVE_FLUID_PVTFUNCTIONS_PVTFUNCTIONHELPERS_HPP
+#define GEOSX_CONSTITUTIVE_FLUID_PVTFUNCTIONS_PVTFUNCTIONHELPERS_HPP
+
+namespace geosx
+{
+
+namespace constitutive
+{
+
+namespace PVTProps
+{
+
+class PTTableCoordinates
+{
+public:
+  PTTableCoordinates()
+  { coords.resize( 2 ); }
+
+  localIndex nPressures() const { return coords[0].size(); }
+  localIndex nTemperatures() const { return coords[1].size(); }
+
+  void appendPressure( const real64 & pres ) { coords[0].emplace_back( pres ); }
+  void appendTemperature( const real64 & temp ) { coords[1].emplace_back( temp ); }
+
+  real64 const & getPressure( localIndex i ) const { return coords[0][i]; }
+  real64 const & getTemperature( localIndex i ) const { return coords[1][i]; }
+
+  array1d< array1d< real64 > > const & getCoords() const { return coords; }
+
+private:
+  array1d< array1d< real64 > > coords;
+};
+
+struct PVTFunctionHelpers
+{
+
+  /**
+   * @brief Look for the expectedNames in the names provided by the user
+   * @param[in] inputNames phase or component names provided by the user
+   * @param[in] expectedNames expected names that can be accepted by GEOSX
+   * @return the index of the phase or component that was found
+   */
+  template< typename InputRange, typename ExpectedRange >
+  static localIndex
+  findName( InputRange const & inputNames,
+            ExpectedRange const & expectedNames )
+  {
+    using std::begin;
+    using std::end;
+    auto it = std::find_first_of( begin( inputNames ), end( inputNames ), begin( expectedNames ), end( expectedNames ) );
+    localIndex const id = std::distance( begin( inputNames ), it );
+    return id;
+  }
+
+  static real64
+  initializePropertyTableWithSalinity( string_array const & inputParameters,
+                                       PTTableCoordinates & tableCoords )
+  {
+    real64 TStart = -1.0;
+    real64 TEnd = -1.0;
+    real64 dT = -1.0;
+    real64 PStart = -1.0;
+    real64 PEnd = -1.0;
+    real64 dP = -1.0;
+    real64 salinity = -1.0;
+
+    GEOSX_ERROR_IF( inputParameters.size() != 9, "Invalid property input!" );
+
+    try
+    {
+      PStart = stod( inputParameters[2] );
+      PEnd = stod( inputParameters[3] );
+      dP = stod( inputParameters[4] );
+
+      TStart = stod( inputParameters[5] );
+      TEnd = stod( inputParameters[6] );
+      dT = stod( inputParameters[7] );
+
+      salinity = stod( inputParameters[8] );
+    }
+    catch( const std::invalid_argument & e )
+    {
+      GEOSX_ERROR( "Invalid BrineCO2Density argument:" + string( e.what()) );
+    }
+
+    for( real64 P = PStart; P <= PEnd; P += dP )
+    {
+      tableCoords.appendPressure( P );
+    }
+    for( real64 T = TStart; T <= TEnd; T += dT )
+    {
+      tableCoords.appendTemperature( T );
+    }
+    return salinity;
```

Comment:
Just put this code inside the `try`/`catch` so you can remove the initial declarations and declare your variables `const`.

PS: Obviously you do not require that you have the same number of `P` and `T`.
Why do you store this in a 2d array instead in 2 separate 1d arrays?

Reply:
Yes I agree, I put the code mentioned above inside the `try/catch`, and I removed the initial declarations as you suggested. 

Regarding the PS, you are right that we don't require the same number of `P` and `T`. But this is exactly why this is stored as an `array1d< array1d< real64 > >`, which supports the case of different numbers of pressures and temperatures. Then `TableFunction` converts this into an `ArrayOfArrays` and everything is good.